ParaLink grant application (#88)
ringecosystem/collaboration: grant/projects/paralink.md

# ParaLink

## Project Overview

[ParaLink](https://para.link/) is a cross-chain asset exchange tool built on the Polkadot XCM (Cross-Consensus Messaging) technology stack. The project is dedicated to creating a seamless cross-chain experience, supporting asset interoperability between parachains within the Polkadot ecosystem.

## Team

> - Snoopy1412 ([@snoopy1412](https://github.com/snoopy1412))

Reviewer: @boundless-forest Can we add the members from Itering?

The other members from Itering working on Paralink might not be covered by this grant, but they are also supported in other form such as service provider. It is ok to list it here, only need to make clear this grant is mainly cover UI for @snoopy1412

Reviewer 2: Done

## Team Code Repos

> - [https://github.com/snoopy1412/paralink](https://github.com/snoopy1412/paralink)

Reviewer 2: Do you have permission to submit code to the RingDAO organization? There is already a repo at https://github.com/ringecosystem/paralink-ui. However, would it be better to create a new one under the RingDAO group?

## Development Roadmap

- **Estimated Duration**: 6 weeks
- **Budget**: 2500 USD
- **Wallet Address**: 0x3d6d656c1bf92f7028Ce4C352563E1C363C58ED5

### Tasks

#### Infrastructure and Core Feature Development

1. **User Interface Development**

   - Build the project’s foundational architecture using React, TypeScript, TailwindCSS, and shadcn/ui.
   - Create a user-friendly interface to optimize the user workflow.
   - Implement comprehensive responsive design to ensure compatibility across multiple devices.
   - Integrate TanStack Query for efficient server-side state management.

2. **Web3 Integration**

   - Integrate wagmi, viem, and RainbowKit to enable interactions with EVM-compatible chains.
   - Integrate @polkadot/api and @polkadot/extension-dapp to facilitate interactions with Polkadot chains.
   - Implement intelligent chain type recognition based on user selection, automatically switching to the corresponding wallet login and interaction logic.

> 3. **Smart XCM Design**

Reviewer: Most content in this section including automation and integration looks not in the scope of UI for @snoopy1412 .

I am not sure that wether it will be delivered solely from @snoopy1412 or not. Might be better split to other parts like Reference or Partners sections to reveal the actual state of work and make the scope for the grant more clear?

     - Tailor asset information retrieval to accommodate differences in on-chain asset management modules across various chains.
     - Implement custom handling for **special assets** that cannot be processed automatically, ensuring their support within ParaLink.

   - **Frontend Display Optimization**

     - Handle the display of **Token and Network Logos** on the frontend, prioritizing data acquisition from [Subscan Assets Info](https://github.com/subscan-explorer/assets-info) and using default icons if necessary.
     - Ensure accurate display of asset and network information on the frontend, providing users with clear information.

   - **Cross-Chain Transfer Logic Enhancement**

     - **Implement cross-chain asset transfer logic between EVM chains and Polkadot chains**, enabling users to transfer assets across different types of chains.
     - Manage various potential exceptions during the cross-chain process to ensure asset security.
     - Dynamically adjust the cross-chain transfer logic and process based on automatically detected chain types and asset types.

##### PREPLK Distribution Commitment

> To express our gratitude for RingDao's funding support for this project, especially in the early stages, ParaLink has launched an ERC-20 token on the Darwinia mainnet called [PREPLK (Pre ParaLink Token)](https://explorer.darwinia.network/address/0x7ED13f74FD8AE70db03Ae74666d1B443341D8A41), and has committed to distribute 20% of the total supply to the RingDao community. We hope this will help the RingDao community gain more benefits from the project and develop a long-term relationship with ParaLink.

Reviewer: 20% distribution of PREPLK include the support of this grant and other early support previous to this. (Hope to make this point more clear.)

In this way to reflect the invest from RingDAO by holding the PREPLK tokens for IP and governance.

@boundless-forest

Reviewer 2: Done

# ParaLink

## Project Overview

[ParaLink](https://para.link) is a cross-chain asset exchange hub built on the Polkadot XCM (Cross-Consensus Messaging) technology stack to serve users of parachains in the Polkadot ecosystem. The current version only supports tokens in the Darwinia ecosystem, and the user experience needs improvement. This grant aims to redesign and rebuild ParaLink, from the user interface to the underlying logic, to make it more user-friendly and efficient.

## Team

- [Snoopy1412](https://github.com/snoopy1412)
- [Bear Wang](https://github.com/boundless-forest)
- [Guantong](https://github.com/jiguantong)

## Team Code Repos

- https://github.com/ringecosystem/paralink

> ## Development Roadmap
> 
> - **Estimated Duration**: 6 weeks
> - **Budget**: 2500 USD
> - **Wallet Address**: 0x3d6d656c1bf92f7028Ce4C352563E1C363C58ED5

Reviewer: ```suggestion
## Grant Information

- **Key Contributor**: Snoopy1412
- **Estimated Duration**: 6 weeks
- **Budget**: 2500 USD
- **Payee Wallet Address**: 0x3d6d656c1bf92f7028Ce4C352563E1C363C58ED5
```

# ParaLink

## Project Overview

[ParaLink](https://para.link) is a cross-chain asset exchange hub built on the Polkadot XCM (Cross-Consensus Messaging) technology stack to serve users of parachains in the Polkadot ecosystem. The current version only supports tokens in the Darwinia ecosystem, and the user experience needs improvement. This grant aims to redesign and rebuild ParaLink, from the user interface to the underlying logic, to make it more user-friendly and efficient.

## Team

- [Snoopy1412](https://github.com/snoopy1412)

> - [Bear Wang](https://github.com/boundless-forest)
> - [Guantong](https://github.com/jiguantong)

Reviewer: ```suggestion
- [Bear Wang](https://github.com/boundless-forest) from Itering
- [Guantong](https://github.com/jiguantong)  from Itering
- [Yuqi] from Itering
```

     - Ensure accurate display of asset and network information on the frontend, providing users with clear information.

   - **Cross-Chain Transfer Logic Enhancement**

     - **Implement cross-chain asset transfer logic between EVM chains and Polkadot chains**, enabling users to transfer assets across different types of chains.
     - Manage various potential exceptions during the cross-chain process to ensure asset security.
     - Dynamically adjust the cross-chain transfer logic and process based on automatically detected chain types and asset types.

## PREPLK Distribution Commitment

To express our gratitude for RingDao's funding support for this grant and the previous research, design and development work, ParaLink has launched an ERC-20 token on the Darwinia mainnet called [PREPLK (Pre ParaLink Token)](https://explorer.darwinia.network/address/0x7ED13f74FD8AE70db03Ae74666d1B443341D8A41), and has committed to distribute 20% of the total supply to the RingDao community. We hope that ParaLink
would be a valuable part of the RingDAO ecosystem in the future and develop a long-term relationship with ParaLink.

## Additional Information

> Super thanks to the RingDAO community members, especially Bear Wang, Guantong and Jay, for their technical support and feedback on the design and the previous of ParaLink. We are excited to continue working to build a better ParaLink. 

Reviewer: ```suggestion
Please note that Paralink is a cross-team collaboration project, with other major contributions coming from Itering, including Bear Wang, Guantong, Jay, Yuqi, and others. Since Itering’s contributions are provided separately as a Service Provider, the grant amount does not include compensation for these efforts. The entirety of this grant’s rewards will be allocated to snoopy1412 for completing the grant scope, with Itering providing the necessary support.
```

     - Tailor asset information retrieval to accommodate differences in on-chain asset management modules across various chains.
     - Implement custom handling for **special assets** that cannot be processed automatically, ensuring their support within ParaLink.

   - **Frontend Display Optimization**

     - Handle the display of **Token and Network Logos** on the frontend, prioritizing data acquisition from [Subscan Assets Info](https://github.com/subscan-explorer/assets-info) and using default icons if necessary.
     - Ensure accurate display of asset and network information on the frontend, providing users with clear information.

   - **Cross-Chain Transfer Logic Enhancement**

     - **Implement cross-chain asset transfer logic between EVM chains and Polkadot chains**, enabling users to transfer assets across different types of chains.
     - Manage various potential exceptions during the cross-chain process to ensure asset security.
     - Dynamically adjust the cross-chain transfer logic and process based on automatically detected chain types and asset types.

## PREPLK Distribution Commitment

> To express our gratitude for RingDao's funding support for this grant and the previous research, design and development work, ParaLink has launched an ERC-20 token on the Darwinia mainnet called [PREPLK (Pre ParaLink Token)](https://explorer.darwinia.network/address/0x7ED13f74FD8AE70db03Ae74666d1B443341D8A41), and has committed to distribute 20% of the total supply to the RingDao community. We hope that ParaLink
> would be a valuable part of the RingDAO ecosystem in the future and develop a long-term relationship with ParaLink.

Reviewer: ```suggestion
To express our gratitude for RingDao's funding support for this grant and the previous research, design and development work, ParaLink has launched an ERC-20 token on the Darwinia mainnet called [PREPLK (Pre ParaLink Token)](https://explorer.darwinia.network/address/0x7ED13f74FD8AE70db03Ae74666d1B443341D8A41), and has committed to distribute 20% of the total supply to the RingDAO Ecosystem Multisig after the grant finish deliver. We hope that ParaLink would be a valuable part of the RingDAO ecosystem in the future and develop a long-term relationship with ParaLink.
```
